src/mscorlib/shared/System/Span.NonGeneric.cs

```diff
@@ -26,6 +26,76 @@ public static class Span
         // s_invariantMode is defined for the perf reason as accessing the instance field is faster than access the static property GlobalizationMode.Invariant
         private static readonly bool s_invariantMode = GlobalizationMode.Invariant;
```

Comment:
This cache does not make sense. `GlobalizationMode.Invariant` property should have the exact same perf as you local cache.

Reply:
This was borrowed from CompareInfo:
https://github.com/dotnet/coreclr/blob/34083faf9352151b1bd007707217f7f22e6b365b/src/mscorlib/shared/System/Globalization/CompareInfo.cs#L77

Should it be removed from there too?

Would calling the Invariant property result in this function call every time?
https://github.com/dotnet/coreclr/blob/b60c91cdb9de36d2356ee8f135de65473f282209/src/mscorlib/src/System/Globalization/GlobalizationMode.cs#L14

Reply:
> Would calling the Invariant property result in this function call every time?

No, the syntax:
```C#
internal static bool Invariant { get; } = GetGlobalizationInvariantMode();
```
is equivalent to:
```C#
private readonly static bool backingFieldInvariant = GetGlobalizationInvariantMode();
internal static bool Invariant => backingFieldInvariant;
```

Reply:
> This was borrowed from CompareInfo:

The one in CompareInfo is instance field. This one is static field. The performance characteristics of instance and static fields are quite different (instance fields are faster).

Reply:
> Should it be removed from there too?

No, in CompareInfo it is an instance field and not static.

Reply:
> The performance characteristics of instance and static fields are quite different (instance fields are faster).

Good to know. I missed the detail of it being an instance field vs. a static field here.

> No, the syntax... is equivalent to...

Ah I see. Thanks for clarifying.